libraries/Microsoft.Bot.Builder/IUserTokenProvider.cs

```diff
@@ -4,13 +4,24 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
 using Microsoft.Bot.Schema;
 
 namespace Microsoft.Bot.Builder
 {
     public interface IUserTokenProvider
     {
-        /// <summary>Attempts to retrieve the token for a user that's in a login flow.
+        /// <summary>Attempts to retrieve the token for a user that's in a login flow, using customized AppCredentials.
+        /// </summary>
+        /// <param name="turnContext">Context for the current turn of conversation with the user.</param>
+        /// <param name="oAuthAppCredentials">AppCredentials for OAuth.</param>
+        /// <param name="connectionName">Name of the auth connection to use.</param>
+        /// <param name="magicCode">(Optional) Optional user entered code to validate.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>Token Response.</returns>
+        Task<TokenResponse> GetUserTokenAsync(ITurnContext turnContext, AppCredentials oAuthAppCredentials, string connectionName, string magicCode, CancellationToken cancellationToken);
```

Comment:
>GetUserTokenAsync [](start = 28, length = 17)

Technically all of these additions are breaking changes to this interface. You should check with ChrisM and Carlos about whether they think this is ok, or if it should be a new interface.

Reply:
Yes Carlos is good with these changes and Chris said he'll rely on Carlos :)